### PR TITLE
Fix: Showing a dialog when opens a deep link with no logged in account.

### PR DESCRIPTION
**Pull request by billypchan**
### Issues

If a user opens a deep link with no logged in account, no info dialog is shown.

### Causes

The pending `URLAction` would not be executed in this case since there is no active user session.

### Solutions

When the app becomes active, AppRootViewController checks if the `authenticationState` is not `loggedIn` and having unhandled deep link stored, show a warning dialog.

### Dependencies
- [x] https://github.com/wireapp/wire-ios-sync-engine/pull/992
- [x] Wait for the text copy of the dialog.

### CODE REVIEW

File: Cartfile.resolved [hunk 1]
```diff
@@ -23,7 +23,7 @@ github "wireapp/wire-ios-link-preview" "17.0.0"
 github "wireapp/wire-ios-protos" "19.0.0"
 github "wireapp/wire-ios-request-strategy" "140.0.1"
 github "wireapp/wire-ios-share-engine" "129.0.0"
-github "wireapp/wire-ios-sync-engine" "241.0.1"
+github "wireapp/wire-ios-sync-engine" "243.0.1"
```
Comment (typfel) [inline Cartfile.resolved]: This doesn't match the `Cartfile`, when bumping a major version you need to update major version in the `Cartfile` and run `carthage update`

File: Wire-iOS/Sources/AppRootViewController.swift [hunk 2]
```diff
@@ -708,3 +700,14 @@ extension AppRootViewController: SessionManagerURLHandlerDelegate {
 extension Notification.Name {
     static let companyLoginDidFinish = Notification.Name("Wire.CompanyLoginDidFinish")
 }
+
+extension UIViewController {
```
Comment (typfel) [inline Wire-iOS/Sources/AppRootViewController.swift]: This change is not necessary, let's create a generic `presentAlert()` method when we have need for it.

Comment (billypchan) [reply Wire-iOS/Sources/AppRootViewController.swift]: @typfel There are several places in the project creating a simple alert with an OK button. I will clean them up.

Comment (typfel) [reply Wire-iOS/Sources/AppRootViewController.swift]: Very well but do in a separate PR

File: Cartfile [hunk 3]
```diff
@@ -1,6 +1,6 @@
 github "wireapp/wire-ios-ziphy" ~> 12.0
 github "wireapp/Cartography" "3.0.3"
-github "wireapp/wire-ios-sync-engine" ~> 240.0
+github "wireapp/wire-ios-sync-engine" ~> 243.0.1
```
Comment (billypchan) [inline Cartfile]: @typfel Cartfile is updated

Comment (typfel) [reply Cartfile]: 👍 No need specify minor & patch version in the `Cartfile` but it also doesn't hurt so it's fine to keep it.

File: Wire-iOS/Sources/AppRootViewController.swift (hunk 2 above)
Comment (typfel) [reply Wire-iOS/Sources/AppRootViewController.swift]: But let's revert this change for now.

**Pull request merged**